src/crate/client/test_result.py

```diff
@@ -0,0 +1,48 @@
+import sys
+import unittest
+
+from crate import client
+from crate.client.test_support import setUpCrateLayerBaseline, tearDownDropEntitiesBaseline
+from crate.testing.settings import crate_host
+
+
+class BulkOperationTest(unittest.TestCase):
+
+    def setUp(self):
+        setUpCrateLayerBaseline(self)
+
+    def tearDown(self):
+        tearDownDropEntitiesBaseline(self)
+
+    @unittest.skipIf(sys.version_info < (3, 8), "BulkResponse needs Python 3.8 or higher")
+    def test_executemany_with_bulk_response(self):
+
+        # Import at runtime is on purpose, to permit skipping.
+        from crate.client.result import BulkResponse
+
+        connection = client.connect(crate_host)
+        cursor = connection.cursor()
+
+        # Run SQL DDL and a first valid INSERT operation.
+        cursor.execute("CREATE TABLE foobar (id INTEGER);")
+        _ = cursor.executemany("INSERT INTO foobar (id) VALUES (?)", [(1, ), (2, ), (3, )])
```

Comment:
To really check bulk execution errors, you need for example to add a PK on id, and then try to insert duplicate ids.

Reply:
Thank you very much. I will change the test case correspondingly, to better exercise partially succeeding batches.

Reply:
The test case has been changed like you suggested, and it works perfectly well.

Reply:
nice!